tests/integration/conftest.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
This is excellent, but I also recommend adding test clean up to remove any resources (NHC, etcd, etc.) downloaded during the integration test run. This prevents the tarballs and generated version files from being accidentally committed. Here is the clean up macro I use that runs right before pytest exits:

```python3
def pytest_sessionfinish(session, exitstatus) -> None:
    """Clean up repository after test session has completed."""
    pathlib.Path(ETCD).unlink(missing_ok=True)
    pathlib.Path(NHC).unlink(missing_ok=True)
    pathlib.Path(VERSION).unlink(missing_ok=True)
```